docs/decisions/0003-name-affirmation-signals.rst

```diff
@@ -0,0 +1,38 @@
+3. Using Signals for Name Affirmation
+-------------------------------------
+
+Status
+------
+
+Accepted
+
+Context
+-------
+
+A `VerifiedName` object has a `status` field that should correspond to the status of the associated IDV or proctored exam attempt.
+The value for the field should therefore be affected by any updates to the status of either the IDV or proctored exam attempt. It
+should also be noted that there is not a direct correlation between the statuses for a `VerifiedName` and the statuses
+used by the IDV and proctoring services, so some level of translation has to occur between the services.
+
+Decision
+--------
+
+* Use Django Plugin Signals to send and receive info about IDV and proctored exam attempts for name affirmation to consume.
+
+  * The signals that each service sends should be defined within that service.
+  * Signal receivers will be defined in the name affirmation service.
+* All name affirmation logic will be contained within the service. This includes the translation logic, and when a `VerifiedName` record should be created or updated.
+* This allows the services sending signals to be decoupled from the name affirmation service, and for all name affirmation logic to be contained within the name affirmation service.
+
+
+Consequences
+------------
+
+* IDV and proctoring services will emit signals each time an IDV or exam attempt record is updated.
```

Comment:
It might be worth mentioning that the signals emitted by IDV and proctoring are "receiver agnostic"; that's just a term I made up to be more polite than "dumb signals" 😛 .

This may be a chicken or the egg situation where it's a question of whether the name affirmation receivers are smart because the emitters are dumb or the emitters are dumb because the receiver should be smart, so it might not be the best fit for the `Consequences` section. Maybe it could go in `Decision`.

Reply:
I tried to explain this in the Decision section, let me know if it makes sense